### PR TITLE
Highest liquidity correct assets order

### LINKED ISSUE #727: Highest liquidity pool in the multi network context
Issue: The assets are not in the correct order 
Solution: Order the assets based on the liquidity

### PR DESCRIPTION
**Pull request by bogdanfazakas**
Fixes #727 .

Changes proposed in this PR:

- order assets in highest liquidity component correctly in multi network context

### CODE REVIEW

**Review by KatunaNorbert — APPROVED** [review]
LGTM

File: src/utils/subgraph.ts [hunk 1]
```diff
@@ -426,26 +430,27 @@ export async function getHighestLiquidityDIDs(
...
-    if (fetchedPools.data?.pools?.length === 0) return null
-    for (let i = 0; i < fetchedPools.data.pools.length; i++) {
-      if (!fetchedPools.data.pools[i].datatokenAddress) continue
-      const did = web3.utils
-        .toChecksumAddress(fetchedPools.data.pools[i].datatokenAddress)
-        .replace('0x', 'did:op:')
-      didList.push(did)
-    }
+  }
+  highestLiquidiyAssets.sort((a, b) => a.valueLocked - b.valueLocked).reverse()
```
Comment (claudiaHash) [inline src/utils/subgraph.ts:447]: Seems like this ordering leads to the correct result.

**Review by claudiaHash — APPROVED** [review]
Looks good to me!

File: src/components/pages/Home.tsx [hunk 2]
```diff
@@ -24,11 +24,10 @@ async function getQueryHighest(
   chainIds: number[]
 ): Promise<[SearchQuery, string]> {
   const dids = await getHighestLiquidityDIDs(chainIds)
-
   // TODO: this query needs to adapt to chainIds
```
Comment (kremalicious) [inline src/components/pages/Home.tsx:27]: I think we did that so can remove the TODO

File: src/utils/subgraph.ts [hunk 3]
```diff
@@ -16,6 +16,10 @@ import {
   AssetsFreePrice_dispensers as AssetFreePriceDispenser
 } from '../@types/apollo/AssetsFreePrice'
 import { AssetPreviousOrder } from '../@types/apollo/AssetPreviousOrder'
+import {
+  HighestLiquidiyAssets_pools as HighestLiquidiyAssetsPools,
```
Comment (kremalicious) [inline src/utils/subgraph.ts:20]: not sure where exactly this is coming from but throughout code small typo with a missing `t` like `HighestLiquidiyAssetsPools` which should be `HighestLiquidityAssetsPools `

File: src/utils/subgraph.ts (hunk 1 above)
Comment (kremalicious) [reply src/utils/subgraph.ts:447]: tending towards sneaking in small change towards ordering by `oceanReserve` in here as we're refactoring and touching it. I'm undecided by now but current implementation indeed feels as only being technically correct, whereas the arguments for `oceanReserve` are all good points from user standpoint like @w1kke & @DMats brought up and we generally tend towards user perspective. And realistically it should only lead to noticeable differences between new and very old pools.

File: src/utils/subgraph.ts (hunk 3 above)
Comment (bogdanfazakas) [reply src/utils/subgraph.ts:20]: right missed that 😃 , it comes from the typing generated by the highest liquidity query, I'll update that also

File: src/components/pages/Home.tsx (hunk 2 above)
Comment (bogdanfazakas) [reply src/components/pages/Home.tsx:27]: sure gonna remove that

File: src/utils/subgraph.ts (hunk 1 above)
Comment (bogdanfazakas) [reply src/utils/subgraph.ts:447]: sure we can try ordering by oceanReserve, i tried to reproduce the actual order in our live market where we sort asset by valueLocked see https://github.com/oceanprotocol/market/blob/main/src/utils/subgraph.ts#L127-L128.
I'm gonna push the changes so we can have a look and decide if we stick to order by valueLocked or go with oceanReserve

**Comment by mihaisc** [comment]
`oceanReserve` would make sense. Our ui is kinda of a mess right now

Screenshot 2021-07-14 172018
Screenshot 2021-07-14 172327

So we should add a label to TVL and change Pool Creator Liquidity to Pool Creator Statistics to be consistent

**Comment by bogdanfazakas** [comment]
> `oceanReserve` would make sense. Our ui is kinda of a mess right now
> 
> Screenshot 2021-07-14 172018
> Screenshot 2021-07-14 172327
> 
> So we should add a label to TVL and change Pool Creator Liquidity to Pool Creator Statistics to be consistent

order by oceanReserve should be fine, going to add also the label for TVL and change the Pool Creator Liquidity to Pool Creator Statistics

**Comment by kremalicious** [comment]
Other than points mentioned implementation is perfect but as an enhancement it seems there could still be value in some split-up of this section showing liquidity categorized based on network. Main reason being discoverability, as there currently is much more liquidity in ETH mainnet, pushing down assets of other networks in that section. But can all be handled in a new issue with some initial design thinking

**Comment by mihaisc** [comment]
You can simply select your desired network, not sure if this is a problem. The discoverability issue we can solve after the split, we shouldn't focus on it now

**Comment by bogdanfazakas** [comment]
> Other than points mentioned implementation is perfect but as an enhancement it seems there could still be value in some split-up of this section showing liquidity categorized based on network. Main reason being discoverability, as there currently is much more liquidity in ETH mainnet, pushing down assets of other networks in that section. But can all be handled in a new issue with some initial design thinking

The discoverability issue can really be a problem, but as Mihai mention with this implementation the users can select the desired networks and see the highest liquidity assets from those networks only and since we have the same implementation for the recently publish I would go with this mixed assets and have a new issue in where we decide how we are going to split-up this section or sections from home page

**Pull request merged**